lib/mix/tasks/gh_pages_docs.ex

```diff
@@ -10,36 +10,93 @@ defmodule Mix.Tasks.GhPagesDocs do
   @shortdoc "Update a documentation version on GH Pages."
 
   @spec run(term) :: :ok
-  def run(_args) do
-    version = "v#{Project.config()[:version]}"
-
-    case File.read!("assets/js/versions.js") |> String.contains?(version) do
-      true ->
-        Mix.Task.run("docs")
-        File.cd!("doc")
-        Mix.shell().cmd("git stash")
-
-        Mix.shell().cmd("git checkout gh-pages")
-
-        case File.mkdir("../#{version}") do
-          result when result in [:ok, {:error, :eexist}] ->
-            Mix.shell().cmd("cp -r ./* ../#{version}")
-            File.cd!("..")
-            Mix.shell().cmd("ls")
-            Mix.shell().cmd("git add #{version}/*")
-            Mix.shell().cmd("git commit -m \"Update #{version}\"")
-            Mix.shell().cmd("rm -rf doc")
-            Mix.shell().cmd("git push origin gh-pages")
-
-          _ ->
-            Mix.raise("Cannot create the #{version} directory")
+  def run([version]) do
+    unless String.contains?(File.read!("assets/js/versions.js"), version) do
+      update_versions_js()
+    end
+
+    version =
+      case version do
+        "latest" -> Project.config()[:version]
+        tag -> "v#{tag}"
+      end
+
+    Mix.Task.run("docs")
+    File.cd!("doc")
+    Mix.shell().cmd("git stash")
+
+    # ignore all the changes, the only changes we want to track are in the
+    # doc directory which is listed .gitignore
+
+    Mix.shell().cmd("git checkout gh-pages")
+
+    case File.mkdir("../#{version}") do
+      result when result in [:ok, {:error, :eexist}] ->
+        Mix.shell().cmd("cp -r ./* ../#{version}")
+        File.cd!("..")
+        Mix.shell().cmd("ls")
+        Mix.shell().cmd("git add #{version}/*")
+
+        unless String.contains?(File.read!("assets/js/versions.js"), version) do
+          update_versions_js()
         end
 
+        update_index_html()
+
+        Mix.shell().cmd("git add assets/js/versions.js")
+        Mix.shell().cmd("git add index.html")
+        Mix.shell().cmd("git commit -m \"Add content for #{version}\"")
+        Mix.shell().cmd("rm -rf doc")
+        Mix.shell().cmd("git push origin gh-pages")
+
       _ ->
-        Mix.raise("""
-        The assets/js/versions.js file is out of date.
-        Please add the newest version to the list.
-        """)
+        Mix.raise("Cannot create the #{version} directory")
     end
   end
+
+  defp update_versions_js() do
+    current = Project.config()[:version]
+    {output, 0} = System.cmd("git", ["tag"], [])
+
+    versions = [
+      current
+      | String.split(output, "\n") --
+          [
+            "0.1.0",
+            "0.10.0",
+            "0.9.0",
+            "1.0.0",
+            "1.0.3",
+            "1.0.4",
+            "1.0.5"
+          ]
```

Comment:
Maybe we can simply put a comment explaining that we're just not generating docs for these tags, just so it's faster to understand in the future

Reply:
Sure